Extensions/PathfindingBehavior/pathfindingruntimebehavior.ts

```diff
@@ -295,16 +354,33 @@ namespace gdjs {
       this._searchContext.setObstacles(this._manager);
       this._searchContext.setCellSize(this._cellWidth, this._cellHeight);
       this._searchContext.setStartPosition(owner.getX(), owner.getY());
-      this._searchContext.setObjectSize(
-        owner.getX() - owner.getDrawableX() + this._extraBorder,
-        owner.getY() - owner.getDrawableY() + this._extraBorder,
-        owner.getWidth() -
-          (owner.getX() - owner.getDrawableX()) +
-          this._extraBorder,
-        owner.getHeight() -
-          (owner.getY() - owner.getDrawableY()) +
-          this._extraBorder
-      );
+      if (this._collisionMethod == PathfindingRuntimeBehavior.LEGACY_COLLISION) {
+        this._searchContext.setObjectSize(
+          owner.getX() - owner.getDrawableX() + this._extraBorder,
+          owner.getY() - owner.getDrawableY() + this._extraBorder,
+          owner.getWidth() -
+            (owner.getX() - owner.getDrawableX()) +
+            this._extraBorder,
+          owner.getHeight() -
+            (owner.getY() - owner.getDrawableY()) +
+            this._extraBorder
+        );
+      }
+      else {
+        const copyHitboxes = PathfindingRuntimeBehavior.deepCloneHitboxes(owner.getHitBoxes());
+        for (let pi = 0; pi < copyHitboxes.length; ++pi) {
+          copyHitboxes[pi].move(-owner.getX(), -owner.getY());
+        }
+        this._searchContext.setObjectHitBoxes(copyHitboxes);
```

Comment:
nitpicking: the "deep cloning" might be the responsibility of `setObjectHitBoxes`?

Reply:
I don't know because we give a shape so we have to center it on the origin first. I don't have clear view of how the roles should be dispatched. I think I will think of a global refactor. May be in a following PR ?